File: shared/schema/events.py
from enum import Enum
from typing import Optional, Any
from pydantic import BaseModel
from datetime import datetime

class EventType(str, Enum):
    temp = "temp"
    timer = "timer"
    corrective_action = "corrective_action"
    alert = "alert"

class Severity(str, Enum):
    low = "low"
    medium = "medium"
    high = "high"
    critical = "critical"

class FoodSafetyEvent(BaseModel):
    id: Optional[str] = None
    device_id: str
    site_id: str
    station: str
    probe_id: str
    type: EventType
    value: Any
    ts: datetime
    payload: Optional[dict] = None

File: shared/schema/events.ts
export type EventType = 'temp' | 'timer' | 'corrective_action' | 'alert';
export type Severity = 'low' | 'medium' | 'high' | 'critical';

export interface FoodSafetyEvent {
  id?: string;
  device_id: string;
  site_id: string;
  station: string;
  probe_id: string;
  type: EventType;
  value: number | string;
  ts: string; // ISO 8601
  payload?: {
    action_taken?: string;
    root_cause?: string;
    disposition?: string;
    severity?: Severity;
    narrative?: string;
    threshold?: number;
    rule_id?: string;
    batch_label?: string;
    hold_minutes?: number;
  };
}
